crossbeam-epoch/src/sync/queue.rs

```diff
@@ -137,17 +134,20 @@ impl<T> Queue<T> {
         let head = self.head.load(Acquire, guard);
         let h = unsafe { head.deref() };
         let next = h.next.load(Acquire, guard);
-        match unsafe { next.as_ref() } {
-            Some(n) if condition(&n.data) => unsafe {
-                self.head
-                    .compare_and_set(head, next, Release, guard)
-                    .map(|_| {
-                        guard.defer_destroy(head);
-                        Some(ManuallyDrop::into_inner(ptr::read(&n.data)))
-                    })
-                    .map_err(|_| ())
-            },
-            None | Some(_) => Ok(None),
+        unsafe {
+            match next.as_ref() {
+                Some(n) if condition(&*n.data.as_ptr()) => {
+                    self.head
+                        .compare_and_set(head, next, Release, guard)
+                        .map(|_| {
+                            guard.defer_destroy(head);
+                            // TODO: Replace with MaybeUninit::read when api is stable
```

Comment:
"when the api is stabilized"?

Reply:
Oh, that can be removed actually, it's a stale comment. Will push a cleanup in a few minutes.